pkg/auth/service.go

```diff
@@ -77,6 +77,14 @@ type EmailInviter interface {
 	InviteUser(ctx context.Context, email string) error
 }
 
+type ExternalPrincipalsService interface {
```

Comment:
Kindly add a comment to the entire interface. The methods themselves are self-explanatory.